### PR TITLE
Added Update Execution cli

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: Prafulla Mahindrakar <prafulla.mahindrakar@gmail.com>

# TL;DR
Refer to testing details from this PR https://github.com/flyteorg/flyteidl/pull/248

Unit tests pending

## Type
- [ ] Bug Fix
- [X] Feature
- [ ] Plugin

## Are all requirements met?

- [ ] Code completed
- [ ] Smoke tested
- [ ] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2045

## Follow-up issue
_NA_

### CODE REVIEW

File: cmd/update/execution.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	cmdCore "github.com/flyteorg/flytectl/cmd/core"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/flyteorg/flytestdlib/logger"
+)
+
+const (
+	updateExecutionShort = "Update execution status"
+	updateExecutionLong  = `
+Activating execution unhides it from cli and UI:
```
Comment (katrogan) [inline cmd/update/execution.go:19]:
```suggestion
Activating an execution shows it in the cli and UI:
```

File: cmd/update/execution.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+func updateExecutionFunc(ctx context.Context, args []string, cmdCtx cmdCore.CommandContext) error {
+	project := config.GetConfig().Project
+	domain := config.GetConfig().Domain
+	if len(args) != 1 {
+		return fmt.Errorf(clierrors.ErrExecutionNotPassed)
+	}
+	executionName := args[0]
+	activateLP := execution.UConfig.Activate
+	archiveLP := execution.UConfig.Archive
+	if activateLP == archiveLP && archiveLP {
```
Comment (katrogan) [inline cmd/update/execution.go:43]: nit: maybe `if activateLP && archiveLP {` for readability?

File: cmd/update/execution.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+				Name:    executionName,
+			},
+			Status: &admin.ExecutionStatus{State: executionState},
+		})
+		if err != nil {
+			fmt.Printf(clierrors.ErrFailedExecutionUpdate, executionName, err)
+			return err
+		}
+	}
+	fmt.Printf("updated execution successfully on %v to state %v", executionName, executionState)
```
Comment (katrogan) [inline cmd/update/execution.go:70]:
```suggestion
	fmt.Printf("updated execution %s successfully to state %s", executionName, executionState.String())
```

**Review by katrogan — DISMISSED** [review]
thank you and sorry for the nit-picks!

File: cmd/update/execution.go (hunk 2 above)
Comment (pmahindrakar-oss) [reply cmd/update/execution.go]: Done

**Review by evalsocket — APPROVED** [review]

File: go.mod [hunk 4]
```diff
@@ -55,3 +55,5 @@ require (
 	k8s.io/client-go v0.21.3
 	sigs.k8s.io/yaml v1.2.0
 )
+
+replace github.com/flyteorg/flyteidl => github.com/flyteorg/flyteidl v0.21.20-0.20220111070000-bdd241a81330
```
Comment (katrogan) [inline go.mod:59]: doesn't this need to be updated?

Comment (pmahindrakar-oss) [reply go.mod:59]: Awaiting on the IDL merge to send an updated PR for this

**Pull request merged**